core/include/tpm2.h

```diff
@@ -0,0 +1,138 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, Linaro Limited
+ *
+ */
+
+#ifndef __TPM2_H__
+#define __TPM2_H__
+
+#include <stdint.h>
+#include <types_ext.h>
+#include <util.h>
+
+#define TPM2_REG_SIZE 0x5000
+
+#define TPM2_HDR_LEN 10
+
+#define TPM2_ACCESS(l) SHIFT_U32(l, 12)
+#define TPM2_INT_ENABLE(l) (SHIFT_U32(l, 12) | BIT(3))
+#define TPM2_STS(l) (SHIFT_U32(l, 12) | SHIFT_U32(3, 3))
+#define TPM2_DATA_FIFO(l) (SHIFT_U32(l, 12) | SHIFT_U32(9, 2))
+#define TPM2_DID_VID(l) (SHIFT_U32(l, 12) | SHIFT_U32(0xF, 8))
+#define TPM2_RID(l) (SHIFT_U32(l, 12) | SHIFT_U32(0x3C1, 2))
+
+enum tpm2_int_flags {
+	TPM2_INT_DATA_AVAIL_INT = BIT(0),
+	TPM2_INT_STS_VALID_INT = BIT(1),
+	TPM2_INT_LOCALITY_CHANGE_INT = BIT(2),
+	TPM2_INT_CMD_READY_INT = BIT(7),
+	TPM2_GLOBAL_INT_ENABLE = BIT(31),
+};
+
+enum tpm2_result {
```

Comment:
Does these values follow some standard? If so please add a reference.

Reply:
Yea they follow https://trustedcomputinggroup.org/wp-content/uploads/TCG_PCClientTPMInterfaceSpecification_TIS__1-3_27_03212013.pdf and specifically "5.9.1 Interrupt Enable"